src/NoteTrack.cpp

```diff
@@ -213,6 +213,21 @@ double NoteTrack::GetEndTime() const
    return GetStartTime() + GetSeq().get_real_dur();
 }
 
+void NoteTrack::DoOnProjectTempoChange(
+   const std::optional<double>& oldTempo, double newTempo)
+{
```

Comment:
Pleased to see you figured this out.  I had forgotten the details of using the allegro library.

So you have completely implemented this new virtual for all except the WaveTracks!

Reply:
Yes - which leaves Audacity in an inconsistent state : change project tempo and all tracks shrink or expand except for the most important one :D

Reply:
Good to start testing.  I think copy and paste will work fine for MIDI and label tracks.  The TrackList listener was all you needed.  (TimeTrack does not copy and paste.)